src/lib/decodeMappingKey.ts

```diff
@@ -0,0 +1,109 @@
+/*
+    This file is part of @erc725/erc725.js.
+    @erc725/erc725.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    @erc725/erc725.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with @erc725/erc725.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/**
+ * @file lib/decodeMappingKey.ts
+ * @author Samuel Videau <@samuel-videau>
+ * @date 2022
+ */
+
+import { isHex } from 'web3-utils';
+import { decodeValueType } from './encoder';
+import { ERC725JSONSchema } from '../types/ERC725JSONSchema';
+import { DynamicKeyPart } from '../types/dynamicKeys';
+
+function make32BytesLong(s: string): string {
+  return new Array(65 - s.length).join('0').concat(s);

```

Comment:
can you use  this instead? [web3.utils.padLeft ](https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#padleft)

Reply:
@CallumGrindle done